pyanaconda/modules/payloads/payload/dnf/dnf.py

```diff
@@ -43,16 +47,40 @@ def type(self):
     @property
     def supported_source_types(self):
         """Get list of sources supported by DNF module."""
-        # TODO: Add supported sources when implemented
-        return None
+        return [
+            SourceType.CDROM,
+            SourceType.HDD,
+            SourceType.HMC,
+            SourceType.NFS,
+            SourceType.REPO_FILES,
+            SourceType.URL
+        ]
 
     def process_kickstart(self, data):
         """Process the kickstart data."""
-        pass
+        source_types = SourceFactory.get_type_for_kickstart(data)
```

Comment:
I don't think this is working. Let's move the method `get_type_for_kickstart` here, because it doesn't really make sense to have it in the factory since only the DNF module is interested in the RPM kickstart sources. It will help with issues regarding mixing sources of different payloads or supporting multiple sources that would come up later.

Reply:
I had the same idea, however, I wanted to keep the same solution we have even now.

I agree that this should be changed. It's unnecessarily complicated solution.

Reply:
I've tried a new solution. See the changed version.